src/opera/pge/dswx_ni/dswx_ni_pge.py

```diff
@@ -60,6 +61,28 @@ class DSWxNIPostProcessorMixin(DSWxS1PostProcessorMixin):
     _post_mixin_name = "DSWxNIPostProcessorMixin"
     _cached_core_filename = None
 
+    def _validate_output_product_filenames(self):
+        """
+        Test method to verify the regular expression used to
+        validate the output product file names assigned by the SAS
+        This method will validate that the filename has the acceptable name
+        through a regular expression.  If the pattern does not match
+        """
+        validated_product_filenames = []
+        pattern = re.compile(
+            r'(?P<project>OPERA)_(?P<level>L3)_(?P<product_type>DSWx)-(?P<source>NI)_(?P<tile_id>T[^\W_]{5})_'
+            r'(?P<acquisition_ts>\d{8}T\d{6}Z)_(?P<creation_ts>\d{8}T\d{6}Z)_(?P<sensor>LSAR)_(?P<spacing>30)_'
+            r'(?P<product_version>v\d+[.]\d+)(_(?P<band_index>B\d{2})_'
+            r'(?P<band_name>WTR|BWTR|CONF|DIAG)|_BROWSE)?[.](?P<ext>tif|tiff|png)$')
+
+        for output_file in self.runconfig.get_output_product_filenames():
+            if pattern.match(output_file.split('/')[-1]):
+                validated_product_filenames.append(output_file)
```

Comment:
I don't think its necessary to compile a list of the validated file names. Since we'll throw an exception if any one file name fails, the list will always end up being equal to the set of all filenames anyhow. So we can simplify to a single conditional like so:

```python
if not pattern.match(os.path.basename(output_file)):  # basename is more intuitive to read than .split('/')[-1]
    error_msg = (...)
    self.logger.critical(self.name, ErrorCode.INVALID_OUTPUT, error_msg)
```